public/app/plugins/datasource/tempo/language_provider.ts

```diff
@@ -0,0 +1,103 @@
+import { HistoryItem, LanguageProvider, SelectableValue } from '@grafana/data';
+import { CompletionItemGroup, TypeaheadInput, TypeaheadOutput } from '@grafana/ui';
+import { Value } from 'slate';
+import { TempoDatasource } from './datasource';
+
+export default class TempoLanguageProvider extends LanguageProvider {
+  datasource: TempoDatasource;
+  tags?: string[];
+  constructor(datasource: TempoDatasource, initialValues?: any) {
+    super();
+
+    this.datasource = datasource;
+    Object.assign(this, initialValues);
+  }
+
+  request = async (url: string, defaultValue: any, params = {}) => {
+    try {
+      const res = await this.datasource.metadataRequest(url, params);
+      return res.data;
+    } catch (error) {
+      console.error(error);
```

Comment:
Shouldn't we propagate the error somewhere?

Reply:
Looking at other language providers, they also just log the error with a TODO comment for better error handling. 

Do you know how the data source error handling works or another datasource that does it well? Ideally I can throw the error and it'll show an error banner